### PR TITLE
[SYCL] Use `sycl/detail/core.hpp` in `test-e2e/UserDefinedReductions/*`

### PR DESCRIPTION
**Pull request by aelovikov-intel**
Continuation of changes started in https://github.com/intel/llvm/pull/12890.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
Depends on https://github.com/intel/llvm/pull/13839 (somehow) and has to be merged after it.

File: sycl/include/sycl/ext/oneapi/experimental/user_defined_reductions.hpp [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #pragma once
 
 #include <sycl/detail/defines.hpp>
+#include <sycl/ext/oneapi/experimental/group_helpers_sorters.hpp>
```
Comment (againull) [inline sycl/include/sycl/ext/oneapi/experimental/user_defined_reductions.hpp:12]: I'm most likely missing something, why we need to include this header for sorting algorithms into reductions header?

Comment (aelovikov-intel) [reply sycl/include/sycl/ext/oneapi/experimental/user_defined_reductions.hpp:12]: `group_with_scratchpad` is defined in the sorting extension, UDR extension depends on that one for it.

Comment (againull) [reply sycl/include/sycl/ext/oneapi/experimental/user_defined_reductions.hpp:12]: I see, thanks!

**Review by againull — APPROVED** [review]

**Review by bso-intel — APPROVED** [review]

**Pull request merged**